book/src/requirements.md

```diff
@@ -0,0 +1,166 @@
+
+Functional protocol requirements are scoped to the protocol design itself, which excludes important broader considerations for Orchard as a whole, such as usability goals for a wallet which aren't directly constrained by the protocol.
+
+Finally, there is a category of proposed requirements that could have been coherently in-scope for Orchard which were excluded due to time & complexity constraints.
+
+These scope clarifications exlude some draft requirements found in earlier revisions of this document, in pull request comments, or elsewhere. Many of those could become ZIPs in their own right.
+
+### ProtoR1. Orchard Relies on a Zero-Knowledge Proving System that is Secure in The Uniform Random String Model
+
+**Requirement:** The Orchard shielded transfer circuit uses a Zero-Knowledge Proving System that is secure in the Uniform Random String (URS) Model, and therefore has no reliance on a trusted setup.
+
+**Rationale - UX:** Users who rely on Orchard are not directly vulnerable to the class of trusted parameter compromises.
+
+**Rationale - Strategic:** As Sprout and Sapling usage declines or are deprecated, the ZEC monetary base as a whole will become protected from any previous trusted setup compromise henceforth.
+
+**Rationale - Strategic:** By avoiding a requirement on a trusted setup, Orchard *and future* upgrades no longer need a parameter setup multiparty computation (MPC) process, which improves protocol agility and the ability to respond quickly to security vulnerabilities.
+
+### ProtoR2. Orchard Addresses
+
+**Requirement:** Orchard introduces a new address format with an unambiguous user-facing encoding.
+
+**Rationale - UX:** A new address format allows users to identify which zero-knowledge proof (ZKP) platform their funds rely on.
+
+**Rationale - UX:** A new address format reinforces the fact that users must migrate funds to opt into Orchard.
+
+**Rationale - UX:** A new address format, in combination with the turnstile design, signals to users a boundary for turnstile-based supply integrity protections.
+
+**Rationale - UX:** A new address format, in combination with the turnstile design, signals to users a boundary for privacy-guarantees.
+
+**Counter-rationale - UX:** This complicates the simplified narrative of "z→z has strong privacy" because migrating between pools has a privacy cost.
+
+**Rationale - Engineering:** A new address format simplifies cryptographic design and performance.
+
+### ProtoR3. An Orchard Shielded Pool and Turnstile
+
+**Requirement:** A new shielded pool is introduced with Orchard. ZEC may only enter the Orchard pool from t-addrs or shielded migration transactions, following the same "turnstile" design as for Sprout and Sapling.
+
+**Rationale - UX:** A separate shielded pool clarifies which funds rely on which underlying ZKP/cryptographic platform.
+
+**Rationale - UX:** A separate shielded pool associated with a new address type makes it easier to reason about migration.
```

Comment:
That isn't clear. Suppose for the sake of argument that we introduced a new pool after Orchard, for some reason such as better performance or to mitigate a cryptographic flaw. Suppose further that this didn't require a change to the abstract form of an address (i.e. (d, pk<sub>d</sub>)). What's the argument, in more detail, for how changing the address type would make it easier to reason about migration? I don't see it. (Remember that address types are hidden by the unified address abstraction anyway; they aren't visible to users.)

Reply:
Oh, maybe you mean that *if* you have a new address type, then it's easier to reason about migration when there is a new shielded pool. But that isn't clear either! We could add an address type that supports detection keys, for example, and there would be no reason to create a new pool for it.